### ISSUE TITLE
Code Is not Being tested, testing integration is needed to help me save time

### ISSUE THREAD

**Issue by jtmorrisbytes** [jtmorrisbytes/CatalogApp#5]
I am spending an enormous amount of time on testing every change I make.  it would be great if there was a way for me to automate or semi automate testing my code so I can focus on actually designing the server and api. I am unsure as to the best way to integrate tests into my code at this time. I do know however that I want to try to use Hypothesis and possibly pytest to try to integrate tests. this might require refactoring of all the configuration code that currently exists in the project.

I would prefer not to ship my code with testing included, to reduce file size

Task #1 is to set up the project for testing.

**Comment by jtmorrisbytes** [comment]
Flask provides a way to integrate testing. so far, Im following along at [`flask testing`](http://flask.pocoo.org/docs/1.0/testing/)

**Comment by jtmorrisbytes** [comment]
Managed to configure setup.py and pytest. since I wanted to be able to use pytest and cython together, the only solution I have been able to come up with is to configure cython in pure cython mode and append _cy to the module name unless I decide to go pure python 3 and exclude __init__.py . If cython tries to compile __init__.py, It fails because a required include file is missing at the linker level. I want the maximum performance possible and I do not know how to exclude init.py from being compiled. if I did find a way to maintain as much backwards compatibility as possible while keeping module names untainted then I would hit the holy grail!
